sklearn/metrics/tests/test_dist_metrics.py

```diff
@@ -103,37 +104,73 @@ def test_cdist(metric_param_grid, X, Y):
             D_scipy_cdist = cdist(X, Y, metric, **kwargs)
 
         dm = DistanceMetricInterface.get_metric(metric, **kwargs)
+
+        # DistanceMetric.pairwise must be consistent
+        # on all combinations of format in {sparse, dense}².
```

Comment:
```suggestion
        # DistanceMetric.pairwise must be consistent for all
        # combinations of formats in {sparse, dense}.
```
How about looping over those combinations?

Reply:
Having it unrolled allow identifying in case of failure which combination fails directly.

Do you think we can get this information if looping on combinations?

Reply:
🤔 Fixtures or let it be as is.

Reply:
At the moment, I do not see how one can use fixtures to perform it.
Feel free to do a PR on top of this one.

Reply:
So let's keep it as is.